db/import_column_family_job.cc

```diff
@@ -366,8 +366,32 @@ Status ImportColumnFamilyJob::GetIngestedFileInfo(
     bool bound_set = false;
     if (iter->Valid()) {
       file_to_import->smallest_internal_key.DecodeFrom(iter->key());
-      iter->SeekToLast();
-      file_to_import->largest_internal_key.DecodeFrom(iter->key());
+      Slice largest;
+      if (strcmp(cfd_->ioptions()->table_factory->Name(), "PlainTable") == 0) {
+        // PlainTable iterator does not support SeekToLast().
+        largest = iter->key();
+        for (; iter->Valid(); iter->Next()) {
+          if (cfd_->internal_comparator().Compare(iter->key(), largest) > 0) {
+            largest = iter->key();
+          }
+        }
+        if (!iter->status().ok()) {
+          return iter->status();
+        }
+      } else {
+        iter->SeekToLast();
+        if (!iter->Valid()) {
+          if (iter->status().ok()) {
+            // The file contains at least 1 key since iter is valid after
+            // SeekToFirst().
+            return Status::Corruption("Can not find largest key in sst file");
+          } else {
+            return iter->status();
+          }
+        }
+        largest = iter->key();
+      }
+      file_to_import->largest_internal_key.DecodeFrom(largest);
       bound_set = true;
     }
```

Comment:
I noticed we are missing a status check here. I'll fix it in a follow up PR since it involves updating some existing unit tests.

Reply:
Thank you very much